### PR TITLE
SAAS-469: chips/tags component

### PR DESCRIPTION
**Pull request by fabio-silva**
https://jira.percona.com/browse/SAAS-469

Implementation details:

- Two components: `Chip` and `ChipAreaInputField`. The former can be used as a standalone component (`isRemovable` prop allows a chip to be removed). As for the latter, it's useful to be used within a form (react-final-form);
- Rules for `ChipAreaInputField`: 
    - No duplicate tags allowed (tags are usually unique, and it makes it easier for us to remove a tag without the need to track them using IDs or other hacky methods)
    - No empty tags allowed;
    - `minTags` validator can be used to make sure it's not empty;
    - On form submission, values are saved in a string array on the form (e.g. `{tags: ["tag1", "tag2"]}`);
- Clicking on the container will automatically focus the input.

Screen Recording 2022-01-06 at 11 51 48
Screenshot 2022-01-11 at 11 44 29

Screenshot 2022-01-11 at 11 44 33

### CODE REVIEW

**Comment by atymchuk** [comment]
@fabio-silva can we have an svg clear (cross) icon with tiny lines instead of that bootstrap icon?

File: src/components/Chip/Chip.styles.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { GrafanaTheme } from '@grafana/data';
+import { css } from 'emotion';
+
+export const getStyles = ({ height, spacing, colors }: GrafanaTheme) => ({
+  wrapper: css`
+    height: ${height.sm}px;
+    color: ${colors.text};
+    line-height: ${height.sm}px;
+    display: inline-block;
+    border-radius: calc(${height.sm}px/2);
```
Comment (atymchuk) [inline src/components/Chip/Chip.styles.ts:10]: I would propose:
1/ much less left/right padding; the current padding looks a bit too large;
2/ a lower value for border-radius; the current radius makes it look too roundish

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]
👏👏👏👏

File: src/components/Chip/Chip.styles.ts (hunk 1 above)
Comment (solovevayaroslavna) [reply src/components/Chip/Chip.styles.ts:10]: I agree with Alex, I also visually want to see the left and right padding less than they are now.

Comment (fabio-silva) [reply src/components/Chip/Chip.styles.ts]: Thanks guys. Just applied the changes, please check the last screenshots

File: src/shared/validators/minTags.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Validator } from './validator.types';
+
+export const minTags = (length: number): Validator => (value: string) => {
+  if (value != null && value.length >= length) {
+    return undefined;
+  }
+
+  return `Must contain at least ${length} tags`;
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply src/shared/validators/minTags.ts]: good one!

**Review by tiagomotasantos — APPROVED** [review]

**Comment by fabio-silva** [comment]
> @fabio-silva can we have an svg clear/cross icon with tiny lines instead of that aged font-awesome icon?

Yep, new screenshot added

**Comment by atymchuk** [comment]
> > @fabio-silva can we have an svg clear/cross icon with tiny lines instead of that aged font-awesome icon?
> 
> Yep, new screenshot added

Madonna, it looks so much cleaner! Aaaawesome, thanks!

**Review by atymchuk — APPROVED** [review]

**Pull request merged**